remote_engine_client/src/client.rs

```diff
@@ -534,52 +549,42 @@ impl Stream for ClientReadRecordBatchStream {
-
-                                let compression = match v.compression() {
-                                    arrow_payload::Compression::None => CompressionMethod::None,
-                                    arrow_payload::Compression::Zstd => CompressionMethod::Zstd,
-                                };
-
-                                ipc::decode_record_batches(
-                                    v.record_batches.swap_remove(0),
-                                    compression,
-                                )
-                                .map_err(|e| Box::new(e) as _)
-                                .context(Convert {
-                                    msg: "decode read record batch",
-                                })
-                                .and_then(
-                                    |mut record_batch_vec| {
-                                        ensure!(
-                                            record_batch_vec.len() == 1,
-                                            InvalidRecordBatchNumber {
-                                                batch_num: record_batch_vec.len()
-                                            }
-                                        );
-                                        record_batch_vec
-                                            .swap_remove(0)
-                                            .try_into()
-                                            .map_err(|e| Box::new(e) as _)
-                                            .context(Convert {
-                                                msg: "convert read record batch",
-                                            })
-                                    },
-                                )
+                    Some(v) => match v {
+                        Arrow(v) => {
+                            if v.record_batches.len() != 1 {
+                                return Poll::Ready(Some(
+                                    InvalidRecordBatchNumber {
+                                        batch_num: v.record_batches.len(),
+                                    }
+                                    .fail(),
+                                ));
+                            }
+                            Poll::Ready(Some(convert_arrow_payload(v)))
+                        }
+                        Metrics(v) => {
+                            if v.record_batches.len() != 1 {
+                                return Poll::Ready(Some(
+                                    InvalidRecordBatchNumber {
+                                        batch_num: v.record_batches.len(),
+                                    }
+                                    .fail(),
+                                ));
                             }
-                        };
-                        Poll::Ready(Some(record_batch))
-                    }
+                            let metrics = match convert_arrow_payload(v) {
```

Comment:
Is it necessary to encode the metrics in arrow format? I guess the raw string is enough?